### ISSUE TITLE
J4 com_user - Add option to disallow URLs in Username and Name

### ISSUE THREAD

**Issue by coolcat-creations** [joomla/joomla-cms#39270]
### Is your feature request related to a problem? Please describe.
Despite Captcha on, I get User registrations with valid emails but with scam in the username so that the valid email gets a spam mail from my Website.

### Describe the solution you'd like
I would like to have an option to shorten the name and Username to max. 20 characters (or any custom value) and to add a blacklist for those fields and disallow any URLs in there.

**Comment by Chaosxmk** [comment]
A max character limit wouldn't work thanks to utilities like bit.ly.

**Comment by coolcat-creations** [comment]
That's right but an URL filter would make sense and additional length limit to prohibit long text in those fields

**Comment by brianteeman** [comment]
username is already limited to 30 characters

**Comment by Quy** [comment]
See closed PR #20142 for discussion.

**Comment by coolcat-creations** [comment]
It's a serious security issue for me :-( please bring the PR back @laoneo @GeraintEdwards

**Comment by coolcat-creations** [comment]
> username is already limited to 30 characters

But name isn't- I have lots of russian spam there

**Comment by brianteeman** [comment]
yes it is

https://github.com/joomla/joomla-cms/blob/b53b47c4b3e609dd90d44f96e84cc1b32cab6e04/components/com_users/forms/registration.xml#L11-L17

**Comment by brianteeman** [comment]
As stated by @laoneo in the other PR there are extensions available. I just quickly tested one that was the first match on the JED and it worked fine https://github.com/Spudley/plg_user_restrictusername

**Comment by coolcat-creations** [comment]
Security should be a core setting :-(

**Comment by brianteeman** [comment]
This isnt security.

**Comment by wojtekxtx** [comment]
@brianteeman than what it is if not securiy? OFC it IS security.
